### PR TITLE
fix(TU-15133): Rename `ref` prop to `embedRef` for proper typing

### LINKED ISSUE #653: Typescript: type issue with components ref
It looks like there is some misalignment in the types when passing a `ref` making necessary to suppress the type error in order to proceed and use the components programmatically.

image

### PR DESCRIPTION
**Pull request by mathio**
Changes:
- Resolves #653
    - Components are unable to access `ref` prop directly. Using  `forwardRef` breaks the typings. Since we do not accept any `ForwardedRef` but rather only `MutableRefObject`, we will rename the prop to type it correctly.
- Update NextJS version in demos, add examples for the App Router available from NextJS version >= 13
    - Related issues:
        - https://github.com/Typeform/embed/issues/649
        - https://github.com/Typeform/embed/issues/652
- Update CI to run on Node version 22

----

BREAKING CHANGES: 
- Rename `ref` to `embedRef` in  order to type it properly
- Remove support for node version 16 (end of life 2023-09-11)
- Bumps React version to > 18.

### CODE REVIEW

**Review by leandro-serafim — APPROVED** [review]

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `7d75d6b2c019bcdabf287b9e295bdbbce1809f3f` [here](https://embed.typeform.com/7d75d6b2c019bcdabf287b9e295bdbbce1809f3f/embed.js).

File: packages/embed/e2e/spec/functional/widget.cy.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ describe('Widget', () => {
   testWidget('/', 'server-side rendering')
 })
 
-describe('Widget - Mobile', () => {
+describe.only('Widget - Mobile', () => {
```
Comment (leandro-serafim) [inline packages/embed/e2e/spec/functional/widget.cy.ts:9]:
```suggestion
describe('Widget - Mobile', () => {
```

**Review by leandro-serafim — APPROVED** [review]
Overall, it looks good, just a small comment n the tests, that were added an `only`.

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `b9fbdaf33ab1ef3e40a6fc8c1407a76dc84703fe` [here](https://embed.typeform.com/b9fbdaf33ab1ef3e40a6fc8c1407a76dc84703fe/embed.js).

**Comment by typeform-ops-gha** [comment]
*[BOT]* Preview available with hash `74bce6b298bb0b4087b70ac72f2b9a32c935b2ab` [here](https://embed.typeform.com/74bce6b298bb0b4087b70ac72f2b9a32c935b2ab/embed.js).

**Pull request merged**